src/GpuAsMailboxprocessorDemo/Merge.fs

```diff
@@ -0,0 +1,159 @@
+namespace GraphBLAS.FSharp.Backend.COOMatrix.Utilities
+
+open Brahma.FSharp.OpenCL
+open OpenCL.Net
+open GraphBLAS.FSharp.Backend.Common
+
+type COOMatrix<'a> =
+    {
+        RowCount: int
+        ColumnCount: int
+        Rows: int[]
+        Columns: int[]
+        Values: 'a[]
+    }
+
+[<AutoOpen>]
+module internal Merge =
+    let merge<'a> (gpu:GPU)  =
```

Comment:
```
let merge<'a> gpu = ...

let funсThatUsesMerge1 ... = 
    let merge = merge<int> gpu
    merge ...

let funcThatUsesMerge2 ... = 
    let merge = merge<int> gpu
    merge ...
```
In the above example kernel ```merge``` will be compiled twice despite being the same. So it will be better to delegate kernel cashing to opencl device (or another computational context)

Reply:
Yes, you are right. But I guess, that such a code can be rewritten in the following manner.
```
let merge<'a> gpu = ...

let mergeInt = merge<int> gpu

let funсThatUsesMerge1 ... = 
    mergeInt ...

let funcThatUsesMerge2 ... = 
    mergeInt ...
```

In this case, the kernel will be compiled only once. This allows one to implement an additional layer for caching if required. I'm not sure, that we should implement it because there is a huge number of details. For example, should we assume that cross-device compilation should be cached? Or should we care about the compilation options?

Reply:
But what if `funсThatUsesMerge1` and `funcThatUsesMerge2` are not even in the same file? They could be defined in different parts of a system. In this case we couldn't pass `mergeInt` in closure.

Reply:
But we still can implement an appropriate solution on top of the provided functionality. For example, we can pass ```mergeInt``` as a parameter. Or provide a global cache. I think that caching should not be a part of Brahma.FSharp. Instead, caching should be implemented in user space in the appropriate way (if necessary).

Reply:
Ok, let's allow the user to override the caching behavior in the inheritors of GPU class. Because we still need a context in which to store the cached kernels.

Reply:
The user should design and create this context. Because, for example, in some cases, it may be necessary to create a distributed or cross-GPU cache (if one has a huge number of identical devices).